### DISCUSSION TITLE
Add a parameter to open all external links in a new tab

### DISCUSSION THREAD

**Discussion by MattDelac** [adityatelange/hugo-PaperMod#760] [Ideas]
It would be nice to have an parameter in the `config.toml` where people could choose to open all external links to new tabs by default (by adding `target=_blank"` in the rendered html).

I looked at generic solutions coming with Hugo and it sounds like this should be handle at the theme level.

Please feel free to point me toward the right direction if there is already a solution.

Thanks in advance

**Comment by iDvel** [comment]
try this:
```
[markup]
  defaultMarkdownHandler = 'blackFriday'
  [markup.blackFriday]
    hrefTargetBlank = true
```
---
if you still want to use Goldmark.
create `render-link.html` at `./themes/<你的主题>/layouts/_default/_markup/render-link.html`.
```
<a href="{{ .Destination | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}{{ if strings.HasPrefix .Destination "http" }} target="_blank" rel="noopener"{{ end }}>{{ .Text | safeHTML }}</a>
```

**Comment by MattDelac** [comment]
> try this:
> 
> ```
> [markup]
>   defaultMarkdownHandler = 'blackFriday'
>   [markup.blackFriday]
>     hrefTargetBlank = true
> ```

This worked perfectly. Thanks 🙌

**Reply by lyhsieh** [reply]
Hello, may I ask where should I add these text?

**Reply by MattDelac** [reply]
I put that in the `config.toml`
![image](https://user-images.githubusercontent.com/18557047/151704183-656c3391-259d-4b3e-8dd0-9e8eacfc7052.png)

**Reply by lyhsieh** [reply]
Thanks a lot!
But my config file is .yml file instead of .toml file. 
Is there a solution for config.yml file?

**Reply by Nydid** [reply]
> Thanks a lot! But my config file is .yml file instead of .toml file. Is there a solution for config.yml file?

```
markup:
  defaultMarkdownHandler: blackFriday
  blackFriday:
    hrefTargetBlank: true
```

**Reply by RoneoOrg** [reply]
> Thanks a lot! But my config file is .yml file instead of .toml file. Is there a solution for config.yml file?

Here is a converter Toml - Yaml: https://toolkit.site/format.html

**Comment by parwam** [comment]
Check this solution by the author of the theme himself, as well:
https://adityatelange.in/blog/hugo-external-link-new-tab/
